exercises/binary-search-tree/Example.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class BinarySearchTree<T>
+    where T : IComparable
+{
+    private BinarySearchTree(T value)
+    {
+        Value = value;
+    }
+
+    public T Value { get; }
+
+    public BinarySearchTree<T> Left { get; private set; }
+
+    public BinarySearchTree<T> Right { get; private set; }
+
+    public BinarySearchTree<T> Insert(T value)
+    {
+        if (value.CompareTo(Value) <= 0)
+        {
+            Left = Insert(value, Left);
+        }
+        else
+        {
+            Right = Insert(value, Right);
+        }
+
+        return this;
+    }
+
+    private BinarySearchTree<T> Insert(T value, BinarySearchTree<T> tree)
+    {
+        if (tree == null)
+        {
+            return Singleton(value);
+        }
+
+        return tree.Insert(value);
+    }
+
+    public IEnumerable<T> AsEnumerable()
+    {
+        foreach (var left in Left?.AsEnumerable() ?? Enumerable.Empty<T>())
+        {
+            yield return left;
+        }
+
+        yield return Value;
+        
+        foreach (var right in Right?.AsEnumerable() ?? Enumerable.Empty<T>())
+        {
+            yield return right;
+        }
+    }
+
+    public static BinarySearchTree<T> Singleton(T value)
+    {
+        return new BinarySearchTree<T>(value);
+    }
+
+    public static BinarySearchTree<T> FromEnumerable(IEnumerable<T> values)
```

Comment:
Singleton method is removed :) Furthermore, to make things more idiomatic C#, I've implemented `IEnumerable<int>`, although I only require the `AsEnumerable()` method to exist in the test. That means users can either implement `IEnumerable<int>` and get `AsEnumerable()` for free, or just implement `AsEnumerable()`.